src/alerts_utils.cc

```diff
@@ -360,45 +370,16 @@ alert_load_state (zlistx_t *alerts, const char *path, const char *filename)
     return rv;
 }
 
-/* fty_proto_zpl automatically encloses values in quotes, like this:
-
-description = "Rule changed"
-
-But JSON contains quotes on its own, and ZPL doesn't support escaping them:
-
-description = "{ "key": "Device {{var1}} does not provide expected data. It may be offline or not correctly configured.", "variables": { "var1": "sc21543331131" } }"
-
-However, ZPL doesn't require quotes and if the drop them, the value is still parsed correctly:
-
-description = { "key": "Device {{var1}} does not provide expected data. It may be offline or not correctly configured.", "variables": { "var1": "sc21543331131" } }
-
-This function drops the quotes added by fty_proto_zpl if they are enclosing JSON string.
-*/
-
-void
-s_drop_quotes_around_json (const char *state_file, const char *path, const char *filename)
+// right-pad the string with spaces to specified size
+uint8_t *
+s_pad (const char *src, size_t padded_size)
 {
-    std::ifstream input_stream (state_file);
-    char *real_state_file = zsys_sprintf ("%s/%s", path, filename);
-    std::ofstream output_stream (real_state_file);
-    std::string line;
-    while (std::getline (input_stream, line)) {
-        size_t json_start = line.find ("\"{");
-        if (json_start != std::string::npos) {
-            size_t json_end = line.rfind ("}\"");
-            if (json_end != std::string::npos) {
-                line.replace (json_start, 2, " {");
-                line.replace (json_end, 2, "} ");
-            }
-        }
-        output_stream << line << std::endl;
-    }
+    uint8_t *padded = (uint8_t *) zmalloc ((padded_size) * sizeof (uint8_t));
+    memcpy (padded, src, strlen (src));
+    for (size_t i = strlen (src); i < padded_size; i++)
+        padded[i] = 0x20; // pad with SPACEs
 
-    log_debug ("storing into state file %s", real_state_file);
-    input_stream.close ();
-    zsys_file_delete (state_file);
-    output_stream.close ();
-    zstr_free (&real_state_file);
+    return padded;
```

Comment:
Should we also set `padded[i] = '\0';` here to be safe, where `i == padded_size` (or `==strlen`, technically, whichever is bigger)?

Reply:
no, that should be already \0 because I used zmalloc :-)

Reply:
Well, the last byte of the buffer (which is not a zero-terminated string per your comment below so the point of my original question is moot and the sanity is checked) at `padded[padded_size-1]` is `0x20` after the loop ;)